internal/services/search/search_service_resource.go

```diff
@@ -341,20 +341,39 @@ func resourceSearchServiceUpdate(d *pluginsdk.ResourceData, meta interface{}) er
 	if err != nil {
 		return fmt.Errorf("retrieving %s: %+v", id, err)
 	}
+
 	if resp.Model == nil {
 		return fmt.Errorf("retrieving existing %s: %+v", id, err)
 	}
+
 	model := *resp.Model
-	if model.Properties == nil {
+	if props := model.Properties; props == nil {
 		return fmt.Errorf("retrieving existing %s: `properties` was nil", id)
 	}
 
+	// The service API has changed where it will not allow the updated model to be
+	// passed to the update PATCH call. You must now create a new update payload
+	// object by removing all of the READ-ONLY fields from the model...
+	// (e.g., privateEndpointConnections, provisioningState, sharedPrivateLinkResources,
+	// status and statusDetails)
+	payload := services.SearchService{
+		Id:         model.Id,
+		Identity:   model.Identity,
+		Location:   model.Location,
+		Name:       model.Name,
+		Properties: pointer.To(services.SearchServiceProperties{}),
+		Sku:        model.Sku,
+		Tags:       model.Tags,
+		Type:       model.Type,
```

Comment:
Can we remove these three properties? These fields should (legacy APIs aside) never need to be set in the root of the payload:

```suggestion
		Identity:   model.Identity,
		Location:   model.Location,
		Properties: pointer.To(services.SearchServiceProperties{}),
		Sku:        model.Sku,
		Tags:       model.Tags,
```

Reply:
Fixed.